src/Core/Commander/Providers/IoDriver_JSON.js

```diff
@@ -19,7 +19,21 @@ define('Core/Commander/Providers/IoDriver_JSON', ['Core/Commander/Providers/IoDr
 
     IoDriver_JSON.prototype.read = function(url) {
         return fetch(url).then(function(response) {
-            return response.json();
+            return response.json().then(
+                function(response) {
+                    if(response.status >= 200 && response.status < 300)
+                        return response;
+                    else{
+                        var error = new Error(response.statusText);
+                        error.response = response;
+                        throw error;
+                    }
+                }).then(
+                function(data){
+                    return data;
+                }).catch(function(error) {
+                    console.log('request failed', error);
```

Comment:
OK, I had not understand what you meant before.
I will try to put the catch statement at the end of the chain and test the status code before the json() function.